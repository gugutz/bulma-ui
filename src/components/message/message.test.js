import React from 'react'
import renderer from 'react-test-renderer'
import Message from './message'

describe('<Message /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Message>
          <Message.Header>header</Message.Header>
          <Message.Body>text</Message.Body>
        </Message>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Message size={size}>
            <Message.Header>header</Message.Header>
            <Message.Body>text</Message.Body>
          </Message>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const colors = [
    'dark',
    'primary',
    'link',
    'info',
    'success',
    'warning',
    'danger'
  ]

  for (const color of colors) {
    it('should support color prop with ' + color, () => {
      const tree = renderer
        .create(
          <Message color={color}>
            <Message.Header>header</Message.Header>
            <Message.Body>text</Message.Body>
          </Message>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
