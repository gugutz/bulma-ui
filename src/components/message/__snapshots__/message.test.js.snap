// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> component should render correctly 1`] = `
<article
  className="message"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support color prop with danger 1`] = `
<article
  className="message is-danger"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support color prop with dark 1`] = `
<article
  className="message is-dark"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support color prop with info 1`] = `
<article
  className="message is-info"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support color prop with link 1`] = `
<article
  className="message is-link"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support color prop with primary 1`] = `
<article
  className="message is-primary"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support color prop with success 1`] = `
<article
  className="message is-success"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support color prop with warning 1`] = `
<article
  className="message is-warning"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support size prop with large 1`] = `
<article
  className="message is-large"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support size prop with medium 1`] = `
<article
  className="message is-medium"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support size prop with normal 1`] = `
<article
  className="message"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;

exports[`<Message /> component should support size prop with small 1`] = `
<article
  className="message is-small"
>
  <div
    className="message-header"
  >
    header
  </div>
  <div
    className="message-body"
  >
    text
  </div>
</article>
`;
